src/applications/bmqtool/m_bmqtool_inpututil.cpp

```diff
@@ -149,11 +150,55 @@ void InputUtil::populateProperties(
             BSLS_ASSERT_SAFE(0 == result);
         } break;  // BREAK
 
-        case MessagePropertyType::E_INT: {
+        case MessagePropertyType::E_INT32: {
             BSLA_MAYBE_UNUSED int result =
-                out->setPropertyAsInt32(name, stoi(value));
+                out->setPropertyAsInt32(name, bsl::stoi(value));
             BSLS_ASSERT_SAFE(0 == result);
         } break;  // BREAK
+
+        case MessagePropertyType::E_INT64: {
+            BSLA_MAYBE_UNUSED int result =
+                out->setPropertyAsInt64(name, bsl::stoll(value));
+            BSLS_ASSERT_SAFE(0 == result);
+        } break;  // BREAK
+
+        case MessagePropertyType::E_BOOL: {
+            bool boolValue;
+            bsl::istringstream(value) >> bsl::boolalpha >> boolValue;
+            BSLA_MAYBE_UNUSED int result = out->setPropertyAsBool(name,
+                                                                  boolValue);
+            BSLS_ASSERT_SAFE(0 == result);
+        } break;  // BREAK
+
+        case MessagePropertyType::E_CHAR: {
+            BSLA_MAYBE_UNUSED int result = out->setPropertyAsChar(
+                name,
+                static_cast<char>(bsl::stoi(value, 0, 16)));
```

Comment:
> But since this is a debug tool for developers, I assume they are aware about format and value limits.

For this point, I prefer to reduce cognitive complexity for developers who use the tool. To automate checks since it's easy anyway, and can speed up the development/debug process. There is a difference between 2 paradigms: `the tool might break without telling, so better know how to pass params correctly` and `even if you pass incorrect params, the tool will notify you`.
If you think it's unnecessary for this PR, let's skip these proposals and also other ones which you think might increase the scope of the PR too much.